plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java

```diff
@@ -311,15 +318,21 @@ private void addContainer(IDeploymentConfig dc, String workspaceID, String image
         dc.getContainer(containerName).setImagePullPolicy(IMAGE_PULL_POLICY_IFNOTPRESENT);
 
         ModelNode dcFirstContainer = ((DeploymentConfig)dc).getNode().get("spec").get("template").get("spec").get("containers").get(0);
+        
+        boolean isDevMachine = isDevMachine(exposedPorts);
+        // Run non-dev machine as root user
+        String UID = isDevMachine ? UID_USER : UID_ROOT;
```

Comment:
What if it's not a devMachine and the user is not root? I'm thinking about a [multi-machine workspace] where the non-dev machine default user is not root. For these scenarios wouldn't it possible to retrieve the image default user?